website_attribute_set/controllers/main.py

```diff
@@ -0,0 +1,421 @@
+from odoo.osv import expression
+from odoo.tools import SQL, float_round, groupby, lazy
+
+from odoo.addons.website.controllers.main import QueryURL
+from odoo.addons.website_sale.controllers import main
+
+
+class WebsiteSale(main.WebsiteSale):
+    @route(
+        [
+            "/shop",
+            "/shop/page/<int:page>",
+            '/shop/category/<model("product.public.category"):category>',
+            '/shop/category/<model("product.public.category"):category>/page/<int:page>',
+        ],
+        type="http",
+        auth="public",
+        website=True,
+        sitemap=main.WebsiteSale.sitemap_shop,
+    )
+    def shop(
+        self,
+        page=0,
+        category=None,
+        search="",
+        min_price=0.0,
+        max_price=0.0,
+        ppg=False,
+        **post,
+    ):
+        if not request.website.has_ecommerce_access():
+            return request.redirect("/web/login")
+        try:
+            min_price = float(min_price)
+        except ValueError:
+            min_price = 0
+        try:
+            max_price = float(max_price)
+        except ValueError:
+            max_price = 0
+
+        Category = request.env["product.public.category"]
+        if category:
+            category = Category.search([("id", "=", int(category))], limit=1)
+            if not category or not category.can_access_from_current_website():
+                raise NotFound()
+        else:
+            category = Category
+
+        website = request.env["website"].get_current_website()
+        website_domain = website.website_domain()
+        if ppg:
+            try:
+                ppg = int(ppg)
+                post["ppg"] = ppg
+            except ValueError:
+                ppg = False
+        if not ppg:
+            ppg = website.shop_ppg or 20
+
+        ppr = website.shop_ppr or 4
+
+        gap = website.shop_gap or "16px"
+
+        request_args = request.httprequest.args
+        attrib_list = request_args.getlist("attribute_value")
+        attrib_values = [[int(x) for x in v.split("-")] for v in attrib_list if v]
+        attributes_ids = {v[0] for v in attrib_values}
+        attrib_set = {v[1] for v in attrib_values}
+        if attrib_list:
+            post["attribute_value"] = attrib_list
+
+        # anyalyze the url args to be used in filter and search
+        request_args = request.httprequest.args
+        additional_attrib_list = request_args.getlist("additional_attribute_value")
+        additional_attrib_values = [
+            [x for x in v.split("-", maxsplit=1)] for v in additional_attrib_list if v
+        ]
+        additional_attrib_values = [
+            [int(sublist[0]), sublist[1]] for sublist in additional_attrib_values
+        ]
+        additional_attrib_set = set(
+            (item[0], item[1]) for item in additional_attrib_values
+        )
+        post["additional_attrib_set"] = additional_attrib_set
+        post["additional_attrib_values"] = additional_attrib_values
```

Comment:
```suggestion
        post["additional_attrib_set"] = additional_attrib_set
        post["additional_attrib_values"] = additional_attrib_values
```
IMO, we should get these two values ​​first. If they exist, continue with the next job. Otherwise, we will exit early and return super.

Reply:
not at all, there is no super to continue with

Reply:
I see you are overriding the entire `shop` ([ref](https://github.com/odoo/odoo/blob/f60188fc9d2efc9e961fb4fecad800e529bee041/addons/website_sale/controllers/main.py#L234)) function instead of using super. This can cause quite a few risks such as Loss of Inheritance from the Base Module, Difficulty in Upgrading Odoo, Loss of Extensibility for Other Modules when `odoo` changes the upstream code...
So I think the optimal way to handle this is to return the original feature early if the processing in this overridden function is as expected.

Reply:
indeed, i have encountered a case where a function is declared to return different values ​​in 2 modules ([see](https://github.com/OCA/account-reconcile/pull/740)). that could cause the same error if later another module like `website_attribute_set` also overrides the whole `shop()` function in the future